### PR TITLE
Added an option in init_index to set random seed

### PR DESCRIPTION
**Pull request by bli25broad**
Dear @searchivarius ,

I modified the codes slightly to allow users to set the random seed, instead of the 100 by default.

Best,
Bo

### CODE REVIEW

**Comment by searchivarius** [comment]
Hi @bli25broad thanks! This is up to @yurymalkov to merge this, though.

File: hnswlib/hnswalg.h [hunk 1]
```diff
@@ -32,7 +32,7 @@ namespace hnswlib {
             loadIndex(location, s);
         }
 
-        HierarchicalNSW(SpaceInterface<dist_t> *s, size_t max_elements, size_t M = 16, size_t ef_construction = 200) :
+        HierarchicalNSW(SpaceInterface<dist_t> *s, size_t max_elements, size_t M = 16, size_t ef_construction = 200, size_t random_seed = 0) :
```
Comment (yurymalkov) [inline hnswlib/hnswalg.h]: Can you please make random_seed = 100 by default? This was previously the default value.

**Comment by yurymalkov** [comment]
@bli25broad Many thanks for the pull request! 
Can you please update the default value of the seed to be 100?

**Comment by bli25broad** [comment]
Hi @yurymalkov, sure thing. Done. Please take a look.

**Comment by yurymalkov** [comment]
Hi @bli25broad,
Thank you!

**Pull request merged**